mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -244,9 +244,9 @@ class LinearSVC @Since("2.2.0") (
       }
 
       bcFeaturesStd.destroy(blocking = false)
-      if (state == null) {
+      if (state == null || state.searchFailed) {
         val msg = s"${optimizer.getClass.getName} failed."
-        logError(msg)
+        logWarning(msg)
         throw new SparkException(msg)
```

Comment:
But the msg in the Exception should include the root reason "line search failed." Current msg only including "LBFGS failed."

Reply:
The message that line search failed is logged by breeze.
https://github.com/scalanlp/breeze/blob/4611fe3f4466ac5b1885e9d6288ba27fc9d205ec/math/src/main/scala/breeze/optimize/FirstOrderMinimizer.scala#L78
I think it is a bit verbose that logging at this place.

Reply:
Yes I know. But sometimes people maybe want to catch the exception and get the detail information from exception. It cannot be got from this.

Reply:
Got it. Fixed.